tests/foreman/api/test_syncplan.py

```diff
@@ -74,1024 +78,986 @@ def valid_sync_interval():
-
-    @tier3
-    def test_positive_synchronize_custom_product_daily_recurrence(self):
-        """Create a daily sync plan with current datetime as a sync date,
-        add a custom product and verify the product gets synchronized on
-        the next sync occurrence
-
-        :id: d60e33a0-f75c-498e-9e6f-0a2025295a9d
-
-        :expectedresults: Product is synchronized successfully.
-
-        :CaseLevel: System
-        """
-        delay = 2 * 60
-        product = entities.Product(organization=self.org).create()
-        repo = entities.Repository(product=product).create()
-        start_date = (
-            datetime.utcnow().replace(second=0) - timedelta(days=1) + timedelta(seconds=delay)
-        )
-        # Create and Associate sync plan with product
-        sync_plan = entities.SyncPlan(
-            organization=self.org, enabled=True, interval='daily', sync_date=start_date
-        ).create()
-        sync_plan.add_products(data={'product_ids': [product.id]})
-        # Wait quarter of expected time
-        self.logger.info(
-            'Waiting {} seconds to check product {}'
-            ' was not synced'.format(delay / 4, product.name)
-        )
-        sleep(delay / 4)
-        # Verify product is not synced and doesn't have any content
-        with self.assertRaises(AssertionError):
-            self.validate_task_status(repo.id, max_tries=1)
-        self.validate_repo_content(repo, ['erratum', 'package', 'package_group'], after_sync=False)
-        # Wait the rest of expected time
-        self.logger.info(
-            'Waiting {} seconds to check product {}'
-            ' was synced'.format((delay * 3 / 4), product.name)
-        )
-        sleep(delay * 3 / 4)
-        # Verify product was synced successfully
-        self.validate_task_status(repo.id, repo_backend_id=repo.backend_identifier)
-        self.validate_repo_content(repo, ['erratum', 'package', 'package_group'])
-
-    @tier3
-    def test_positive_synchronize_custom_product_weekly_recurrence(self):
-        """Create a weekly sync plan with a past datetime as a sync date,
-        add a custom product and verify the product gets synchronized on
-        the next sync occurrence
-
-        :id: ef52dd8e-756e-429c-8c30-b3e7db2b6d61
-
-        :expectedresults: Product is synchronized successfully.
-
-        :BZ: 1396647
-
-        :CaseLevel: System
-        """
-        delay = 2 * 60
-        product = entities.Product(organization=self.org).create()
-        repo = entities.Repository(product=product).create()
-        start_date = (
-            datetime.utcnow().replace(second=0) - timedelta(weeks=1) + timedelta(seconds=delay)
-        )
-        # Create and Associate sync plan with product
+    :CaseImportance: Critical
+    """
+    sync_plan = entities.SyncPlan(organization=module_org).create()
+    sync_plan.name = name
+    sync_plan.update(['name'])
+    sync_plan = sync_plan.read()
```

Comment:
I don't think you need to call `.read()` explicitly. What you do need is assigning `.update()` return value to some variable, as it does not modify object in place. Following would do just fine:

```python
    sync_plan = entities.SyncPlan(organization=module_org).create()
    sync_plan.name = name
    sync_plan = sync_plan.update(['name'])
    assert sync_plan.name == name
```

Changing that pattern for all occurrences now would be a lot of work, so since your approach does work, let's leave it as is. But I'm adding this comment for a future.

Reply:
I see, thank you